src/graphics/components/SpriteSheet.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+** EPITECH PROJECT, 2024
+** Stellar-Forge components
+** File description:
+** SpriteSheet.cpp
+**/
+
+#include "SpriteSheet.hpp"
+
+SpriteSheet::SpriteSheet(IObject* owner, const char *path, std::vector<sf::IntRect> frames, int frame) : AGraphicsComponent(owner), path(path), frames(frames), currentFrame(frame)
+{
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+    }
+    sprite.setTexture(texture);
+    sprite.setTextureRect(frames[currentFrame]);
+}
+
+void SpriteSheet::render(sf::RenderWindow *window)
+{
+    for (auto &component : this->getOwner()->getComponents()) {
+        if (component->getMeta().getName() == "Transform") {
+            auto transformComponent = dynamic_cast<Transform *>(component);
+            if (transformComponent != nullptr) {
+                sprite.setPosition(transformComponent->getPosition().x, transformComponent->getPosition().y);
+                sprite.setRotation(transformComponent->getRotation().x);
+                sprite.setScale(transformComponent->getScale().x, transformComponent->getScale().y);
+                sprite.setTextureRect(frames[currentFrame]);
+                window->draw(sprite);
+            }
+        }
+    }
+}
+
+void SpriteSheet::setTexture(const char *path)
+{
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+    }
+    sprite.setTexture(texture);
+}
+
+void SpriteSheet::setFrames(std::vector<sf::IntRect> frames)
+{
+    this->frames = frames;
+}
+
+void SpriteSheet::setFrame(unsigned int frame)
+{
+    currentFrame = frame;
```

Comment:
I think it will be better to check if the frame that they want is correct (no out of range)